### PR TITLE
Log parser refactoring

### PR DESCRIPTION
**Pull request by i-nahornyi**
Hello!
I'd like to contribute to your awesome project by submitting some refactored code. I’ve gone through
**parser/parser.go** and made a few changes to improve its maintainability.

1. **Made the code easier to read:** I cleaned up some parts to make it more straightforward and consistent.
2. **Broke down big chunks of code:** I split up larger functions/classes into smaller, reusable pieces, so it's more modular.
3. **Reduced repetitive code:** I noticed some duplicate logic, so I consolidated it into helper functions.

I’d be really grateful if you could merge this into the main project. But if it’s not a fit right now, no worries—I’ll keep working on it in my fork. Let me know what you think!

P.S.: Go-lang isn’t my main language, so I’m definitely open to any feedback on how to improve this further!

### CODE REVIEW

File: influx/influx.go [hunk 1]
```diff
@@ -45,10 +45,10 @@ type testInfo struct {
 	testStartTime   time.Time
 }
 
-type userLineData struct {
-	timestamp time.Time
-	scenario  string
-	status    string
+type UserLineData struct {
```
Comment (ghost) [inline influx/influx.go:48]: Wondered why the change in capitalization. 

If I understand correcty, capitalization is for exported types and field and non-capitalized is are not exported. Is that correct?

File: parser/errorlineparse.go
Comment (ghost) [inline parser/errorlineparse.go:1]: The filename is missing an 'r': should be `errorlineparser.go`?

I have been reading a bit about [file name conventions](https://stackoverflow.com/questions/25161774/what-are-conventions-for-filenames-in-go) in golang: preferably short. But if filenames are longer I found both suggestions: use underscores or [mixedCaps](https://go.dev/doc/effective_go#mixed-caps). 

Maybe use `error_line_parser.go` or `errorLineParser.go` to make these more readable? Or skip `parser` because it is already the `parser` package.

File: parser/utils.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package parser
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func parseDuration(split [][]byte, startIndex int, endIndex int, entryType string) (int, error) {
```
Comment (ghost) [inline parser/utils.go:10]: In this case the non-capitalized function means not exported outside of package? (also for all other functions in this file and other files)?

**Review by ghost — CHANGES_REQUESTED** [review]
Looks good! Better structured, and less big files. Made some comments and questions.

Note my "custom" languages are Java and Kotlin, so for me golang is also learning steps.

Also note that we moved to the [x2i](https://github.com/perfana/x2i) repository where multiple variants are combined into one (parses Gatling, k6 and jMeter output).

After you have looked and updated the PR, I am happy to merge in into this repo, but we will be archiving this one.

If you would be able to do the similar refactoring in the x2i repo, that would be great! Thanks.

File: influx/influx.go (hunk 1 above)
Comment (i-nahornyi) [reply influx/influx.go:48]: You’re absolutely right! 
I decided to export this type so that it can be reused in `userlineparser.go`. 
Additionally, we can simplify the function signature slightly by changing:

```go
func SendUserLineData(timestamp time.Time, scenario, status string)
```

to:

```go
func SendUserLineData(userLineData UserLineData)
```

File: parser/errorlineparse.go
Comment (i-nahornyi) [reply parser/errorlineparse.go:1]: My mistake (
I’ll add a commit with the naming changes.

File: parser/utils.go (hunk 2 above)
Comment (i-nahornyi) [reply parser/utils.go:10]: Yes, that's correct.

In this case, we don’t need visibility outside the `parser` package because the functionality is only relevant to the parsing logic.

Maybe I should rename the file to `parsing_utils.go`?

**Comment by i-nahornyi** [comment]
@PeterPaul-Perfana 
I've already checked out x2i, and it looks great (like a Swiss Army knife!). 
For now, I plan to finish my current work because I need to stabilize my existing Gatling to InfluxDB integration.

In the near future, I believe I can contribute some things to x2i, so let's stay in touch!

File: parser/utils.go (hunk 2 above)
Comment (ghost) [reply parser/utils.go:10]: Not needed to rename `utils.go`, it is already in the `parser` package.

**Review by ghost — APPROVED** [review]
Thanks for fixes, looks good now.
